Docusaurus `style` prop expects a mapping from style properties to values, not a string

### Bug description

Docusaurus server-side rendering could not render static page with path ... because of error: The `style` prop expects a mapping from style properties to values, not a string. For example, style={{marginRight: spacing + 'em'}} when using JSX.

### Steps to reproduce

_No response_

### Expected behavior

_No response_

### Actual behavior

_No response_

### Your environment

_No response_

### Quarto check output

_No response_

Thanks for trying to help @cscheid ! It happens on deploying to Vercel (which until 22 days ago was deploying fine).. Here's another log from my production app.. let me see I can get a minimal repro.

`[00:05:43.257] Running build in Washington, D.C., USA (East) – iad1
[00:05:43.364] Cloning github.com/krishaamer/green-filter-research (Branch: main, Commit: ed9aaee)
[00:05:43.661] Previous build cache not available
[00:05:45.713] Cloning completed: 2.349s
[00:05:45.969] Running "vercel build"
[00:05:46.413] Vercel CLI 33.7.0
[00:05:47.020] Installing dependencies...
[00:05:47.362] yarn install v1.22.22
[00:05:47.444] [1/4] Resolving packages...
[00:05:47.839] [2/4] Fetching packages...
[00:06:01.107] [3/4] Linking dependencies...
[00:06:01.111] warning "@docusaurus/core > react-loadable-ssr-addon-v5-slorber@1.0.1" has unmet peer dependency "react-loadable@*".
[00:06:01.114] warning "@docusaurus/core > react-dev-utils > fork-ts-checker-webpack-plugin@6.5.3" has unmet peer dependency "typescript@>= 2.7".
[00:06:01.117] warning "@docusaurus/plugin-ideal-image > @slorber/react-ideal-image@0.0.12" has unmet peer dependency "prop-types@>=15".
[00:06:01.119] warning " > @mdx-js/react@3.0.1" has unmet peer dependency "@types/react@>=16".
[00:06:01.119] warning "@docusaurus/preset-classic > @docusaurus/theme-search-algolia > @docsearch/react > @algolia/autocomplete-preset-algolia@1.9.3" has unmet peer dependency "@algolia/client-search@>= 4.9.1 < 6".
[00:06:01.120] warning "@docusaurus/preset-classic > @docusaurus/theme-search-algolia > @docsearch/react > @algolia/autocomplete-core > @algolia/autocomplete-plugin-algolia-insights@1.9.3" has unmet peer dependency "search-insights@>= 1 < 3".
[00:06:01.120] warning "@docusaurus/preset-classic > @docusaurus/theme-search-algolia > @docsearch/react > @algolia/autocomplete-core > @algolia/autocomplete-shared@1.9.3" has unmet peer dependency "@algolia/client-search@>= 4.9.1 < 6".
[00:06:07.369] [4/4] Building fresh packages...
[00:06:08.796] Done in 21.44s.
[00:06:08.833] Running "yarn run build"
[00:06:09.027] yarn run v1.22.22
[00:06:09.058] $ docusaurus build
[00:06:10.332] [INFO] [en] Creating an optimized production build...
[00:06:11.377] [info] [webpackbar] Compiling Client
[00:06:11.399] [info] [webpackbar] Compiling Server
[00:06:34.797] [success] [webpackbar] Server: Compiled successfully in 23.40s
[00:06:44.670] [success] [webpackbar] Client: Compiled successfully in 33.29s
[00:06:48.348] 
[00:06:48.351] [ERROR] Error: Unable to build website for locale en.
[00:06:48.351]     at tryToBuildLocale (/vercel/path0/node_modules/@docusaurus/core/lib/commands/build.js:53:19)
[00:06:48.351]     at async /vercel/path0/node_modules/@docusaurus/core/lib/commands/build.js:64:9
[00:06:48.351]     at async mapAsyncSequential (/vercel/path0/node_modules/@docusaurus/utils/lib/jsUtils.js:20:24)
[00:06:48.351]     at async Command.build (/vercel/path0/node_modules/@docusaurus/core/lib/commands/build.js:62:5) {
[00:06:48.351]   [cause]: Error: Docusaurus static site generation failed for 3 paths:
[00:06:48.351]   - "/research/discussion"
[00:06:48.351]   - "/research/literature/ai"
[00:06:48.352]   - "/research/literature/sustainability"
[00:06:48.352]       at generateStaticFiles (/vercel/path0/node_modules/@docusaurus/core/lib/ssg.js:85:15)
[00:06:48.352]       at async executeSSG (/vercel/path0/node_modules/@docusaurus/core/lib/commands/build.js:146:23)
[00:06:48.352]       ... 4 lines matching cause stack trace ...
[00:06:48.352]       at async Command.build (/vercel/path0/node_modules/@docusaurus/core/lib/commands/build.js:62:5) {
[00:06:48.352]     [cause]: AggregateError
[00:06:48.352]         at generateStaticFiles (/vercel/path0/node_modules/@docusaurus/core/lib/ssg.js:86:20)
[00:06:48.352]         at async executeSSG (/vercel/path0/node_modules/@docusaurus/core/lib/commands/build.js:146:23)
[00:06:48.352]         at async buildLocale (/vercel/path0/node_modules/@docusaurus/core/lib/commands/build.js:126:31)
[00:06:48.352]         at async tryToBuildLocale (/vercel/path0/node_modules/@docusaurus/core/lib/commands/build.js:46:13)
[00:06:48.352]         at async /vercel/path0/node_modules/@docusaurus/core/lib/commands/build.js:64:9
[00:06:48.352]         at async mapAsyncSequential (/vercel/path0/node_modules/@docusaurus/utils/lib/jsUtils.js:20:24)
[00:06:48.352]         at async Command.build (/vercel/path0/node_modules/@docusaurus/core/lib/commands/build.js:62:5) {
[00:06:48.352]       [errors]: [
[00:06:48.352]         Error: Can't render static file for pathname "/research/discussion"
[00:06:48.352]             at generateStaticFile (/vercel/path0/node_modules/@docusaurus/core/lib/ssg.js:119:15)
[00:06:48.352]             at runNextTicks (node:internal/process/task_queues:60:5)
[00:06:48.352]             at process.processImmediate (node:internal/timers:449:9)
[00:06:48.352]             at async /vercel/path0/node_modules/p-map/index.js:57:22 {
[00:06:48.352]           [cause]: Error: Expected component `Figure` to be defined: you likely forgot to import, pass, or provide it.
[00:06:48.352]               at _missingMdxReference (server.bundle.js:1041:9)
[00:06:48.352]               at _createMdxContent (server.bundle.js:575:16)
[00:06:48.352]               at MDXContent (server.bundle.js:1038:8)
[00:06:48.352]               at Uc (server.bundle.js:46586:44)
[00:06:48.352]               at Xc (server.bundle.js:46588:253)
[00:06:48.352]               at Z (server.bundle.js:46594:89)
[00:06:48.352]               at Xc (server.bundle.js:46592:231)
[00:06:48.352]               at Z (server.bundle.js:46594:89)
[00:06:48.352]               at Xc (server.bundle.js:46588:481)
[00:06:48.352]               at Z (server.bundle.js:46594:89)
[00:06:48.353]         },
[00:06:48.353]         Error: Can't render static file for pathname "/research/literature/ai"
[00:06:48.353]             at generateStaticFile (/vercel/path0/node_modules/@docusaurus/core/lib/ssg.js:119:15)
[00:06:48.353]             at runNextTicks (node:internal/process/task_queues:60:5)
[00:06:48.353]             at process.processImmediate (node:internal/timers:449:9)
[00:06:48.353]             at async /vercel/path0/node_modules/p-map/index.js:57:22 {
[00:06:48.353]           [cause]: Error: The `style` prop expects a mapping from style properties to values, not a string. For example, style={{marginRight: spacing + 'em'}} when using JSX.
[00:06:48.353]               at Fa (server.bundle.js:46543:49)
[00:06:48.353]               at K (server.bundle.js:46545:44)
[00:06:48.353]               at Pa (server.bundle.js:46555:222)
[00:06:48.353]               at Xc (server.bundle.js:46589:32)
[00:06:48.353]               at Z (server.bundle.js:46594:89)
[00:06:48.353]               at Yc (server.bundle.js:46597:98)
[00:06:48.353]               at $c (server.bundle.js:46596:140)
[00:06:48.353]               at Z (server.bundle.js:46594:345)
[00:06:48.353]               at Yc (server.bundle.js:46597:98)
[00:06:48.353]               at Xc (server.bundle.js:46589:145)
[00:06:48.353]         },
[00:06:48.353]         Error: Can't render static file for pathname "/research/literature/sustainability"
[00:06:48.354]             at generateStaticFile (/vercel/path0/node_modules/@docusaurus/core/lib/ssg.js:119:15)
[00:06:48.354]             at runNextTicks (node:internal/process/task_queues:60:5)
[00:06:48.354]             at process.processImmediate (node:internal/timers:449:9)
[00:06:48.354]             at async /vercel/path0/node_modules/p-map/index.js:57:22 {
[00:06:48.354]           [cause]: Error: The `style` prop expects a mapping from style properties to values, not a string. For example, style={{marginRight: spacing + 'em'}} when using JSX.
[00:06:48.354]               at Fa (server.bundle.js:46543:49)
[00:06:48.354]               at K (server.bundle.js:46545:44)
[00:06:48.354]               at Ka (server.bundle.js:46548:184)
[00:06:48.354]               at Pa (server.bundle.js:46556:68)
[00:06:48.354]               at Xc (server.bundle.js:46589:32)
[00:06:48.354]               at Z (server.bundle.js:46594:89)
[00:06:48.354]               at Yc (server.bundle.js:46597:98)
[00:06:48.354]               at $c (server.bundle.js:46596:140)
[00:06:48.354]               at Z (server.bundle.js:46594:345)
[00:06:48.354]               at Xc (server.bundle.js:46589:476)
[00:06:48.354]         }
[00:06:48.354]       ]
[00:06:48.354]     }
[00:06:48.354]   }
[00:06:48.354] }
[00:06:48.355] [INFO] Docusaurus version: 3.2.1
[00:06:48.355] Node version: v20.12.1
[00:06:48.421] error Command failed with exit code 1.
[00:06:48.422] info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
[00:06:48.436] Error: Command "yarn run build" exited with 1
[00:06:48.919] `

--- a/news/changelog-1.5.md
+++ b/news/changelog-1.5.md
@@ -25,6 +25,7 @@ All changes included in 1.5:
 ## Docusaurus Format
 
 - ([#8919](https://github.com/quarto-dev/quarto-cli/issues/8919)): Ensure enough backticks in code cell declarations.
+- ([#9179](https://github.com/quarto-dev/quarto-cli/issues/9179)): Emit tables that Pandoc would write as HTML as RawBlock elements to ensure they are rendered correctly in Docusaurus.
 
 ## Website
 

--- a/src/resources/extensions/quarto/docusaurus/docusaurus.lua
+++ b/src/resources/extensions/quarto/docusaurus/docusaurus.lua
@@ -113,6 +113,15 @@ local function tabset(node)
   return tabs
 end
 
+local function Table(tbl)
+  local out = pandoc.write(pandoc.Pandoc({tbl}), FORMAT, PANDOC_WRITER_OPTIONS)
+  -- if the table was written in a way that looks like HTML, then wrap it in the right RawBlock way
+  if string.match(out, "^%s*%<table") then
+    local unwrapped = pandoc.RawBlock('html', out)
+    return RawBlock(unwrapped)
+  end
+end
+
 quarto._quarto.ast.add_renderer("Tabset", function()
   return quarto._quarto.format.isDocusaurusOutput()
 end, function(node)
@@ -178,6 +187,7 @@ return {
     RawBlock = RawBlock,
     DecoratedCodeBlock = DecoratedCodeBlock,
     CodeBlock = CodeBlock,
+    Table = Table,
   },
   {
     Pandoc = Pandoc,

--- a/tests/docs/smoke-all/2024/04/10/9179.qmd
+++ b/tests/docs/smoke-all/2024/04/10/9179.qmd
@@ -1,0 +1,14 @@
+---
+title: Broken Tables
+_quarto:
+  tests:
+    docusaurus-md:
+      ensureFileRegexMatches:
+        - ["dangerouslySetInnerHTML"]
+---
+
+# Broken Markdown Tables
+
++-------+
+| # ad  |
++-------+

